### PR TITLE
[INLONG-5094][SortStandalone]  Implements time interval interceptor

### LINKED ISSUE #5094: [Feature][Sort-Standalone] Implements time interval interceptor
### Description

Interceptor that used to filter data with invalid msg time 

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by vernedeng**
- Fixes #5094

### Motivation

Implements time interval interceptor to support Sort-standalone rollback sender

### Modifications

Implements time interval interceptor

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [x] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

File: inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/rollback/TimeBasedFilterInterceptor.java [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+                .filter(Objects::nonNull)
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public void close() {
+        //no-op
+    }
+
+    public static class Builder implements Interceptor.Builder  {
```
Comment (healchow) [inline inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/rollback/TimeBasedFilterInterceptor.java:89]: Please add Java doc for class, thanks.

File: inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/rollback/TimeBasedFilterInterceptor.java [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    private final long stopTime;
+
+    public TimeBasedFilterInterceptor(long startTime, long stopTime) {
+        this.startTime = startTime;
+        this.stopTime = stopTime;
+    }
+
+    @Override
+    public void initialize() {
+        //no-op
```
[the remark this answers is not in the document]
Comment (vernedeng) [reply inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/rollback/TimeBasedFilterInterceptor.java]: fix, thx

File: inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/rollback/TimeBasedFilterInterceptor.java (hunk 1 above)
Comment (vernedeng) [reply inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/rollback/TimeBasedFilterInterceptor.java:89]: comment has been added, thx

**Review by healchow — APPROVED** [review]
LGTM

**Review by dockerzhang — APPROVED** [review]

**Review by luchunliang — APPROVED** [review]
OK

**Pull request merged**